learn-ocaml.el

```diff
@@ -328,24 +328,48 @@ To be used as a `make-process' sentinel, using args PROC and STRING."
          (list (list learn-ocaml-command-name command token-option server-option id-option html-option dont-submit-option local-option param1 param2)))
     (cl-remove-if-not 'stringp list)))
 
+(cl-defun learn-ocaml-init-user-command-constructor (&key server login password nickname secret)
+  "Construct a shell command with `learn-ocaml-command-name' and options."
+  (let* ((nickname-option (when nickname (concat " " nickname)))
+         (secret-option (when secret (concat " " secret)))
```

Comment:
Thanks @LouisAyroles for your commit.

[IINM](https://en.wiktionary.org/wiki/IINM), I guess that the `concat " "` here are unnecessary (not tested though, I just read the commit diff :)

Indeed, the overall idea of `make-process` and related functions is that instead of launching a shell and evaluating a command like `(concat "learn-ocaml-client" " " "-s" " " "http://localhost:8080" …)` (i.e., a string with space-sep. args),
we run something like `'("learn-ocaml-client" "-s" "http://localhost:8080" …)` (i.e., a list with each program arguments)

Reply:
You're welcome @erikmd
OK, I understand. I think now I've done things pretty much the right way. But I have this error:0 ```error in process sentinel: Keyword argument http://localhost:8080 not one of (:server :login :password :nickname :secret)```
I don't fully understand how we start the process once it's created.
Maybe we can discuss it tomorrow during the daily stand up meeting.